cmd/buildkitd/main.go

```diff
@@ -928,3 +933,20 @@ func (t *traceCollector) Export(ctx context.Context, req *tracev1.ExportTraceSer
 	}
 	return &tracev1.ExportTraceServiceResponse{}, nil
 }
+
+type closeableService interface {
+	Shutdown(ctx context.Context) error
+}
+
+func shutdown(ctx context.Context, closers ...closeableService) error {
+	for _, c := range closers {
+		if c == nil {
+			continue
+		}
+
+		if err := c.Shutdown(ctx); err != nil {
+			return err
+		}
```

Comment:
I feel like this should always shutdown all `closers` [even if one fails] and then return an aggregate error